src/runtime_src/core/tools/xbutil2/xbutil

```diff
@@ -1,19 +1,134 @@
 #!/bin/bash
 
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2020-2021 Xilinx, Inc.
+#
+# xbutil loader which automatically switches between classic and next
+# generation xbutil utility. The current default is classic xbutil tool.
+#
+
 # -- Detect a Windows environment and automatically switch to the .bat file
 if [[ "`uname`" == windows32* ]] || [[ "`uname`" == CYGWIN* ]] ; then
   trap "" INT
   "$0.bat" "$@"
   exit $?
 fi
 
+DEFAULT=xbutil
+XBUTIL=xbutil
+XBUTIL2=xbutil2
+
 # Working variables
 XRT_PROG=xbutil
 
+# Table to resolve validate sub-command based on switch passed to validate
+declare -A VALOPT=( \
+        ["--device"]="XBUTIL2" \
+        ["--format"]="XBUTIL2" \
+        ["--run"]="XBUTIL2" \
+        ["--output"]="XBUTIL2" \
+        ["--help"]="XBUTIL2" \
+        ["-d"]="DEVICE" \
+        ["-f"]="XBUTIL2" \
+        ["-r"]="XBUTIL2" \
+        ["-o"]="XBUTIL2" \
+        ["-h"]="XBUTIL2" \
+    )
+
+# Table to resolve reset sub-command based on switch passed to reset
+declare -A RSTOPT=( \
+        ["--device"]="XBUTIL2" \
+        ["--type"]="XBUTIL2" \
+        ["--help"]="XBUTIL2" \
+        ["-d"]="DEVICE" \
+        ["-r"]="XBUTIL2" \
+        ["-h"]="XBUTIL2" \
+    )
+
+# Table to resolve program sub-command based on switch passed to program
+declare -A PROGOPT=( \
+        ["-p"]="XBUTIL" \
+        ["--user"]="XBUTIL2" \
+        ["-u"]="XBUTIL2" \
+        ["--shell"]="XBUTIL2" \
+        ["-s"]="XBUTIL2" \
+        ["--base"]="XBUTIL2" \
+        ["-b"]="XBUTIL2" \
+        ["--cache"]="XBUTIL2" \
+        ["-c"]="XBUTIL2" \
```

Comment:
Aren't these xbmgmt2 options?

Reply:
Sharp eye! Will update.

Reply:
Will also make the change for reset

Reply:
updated

Reply:
Will you be updating the xbutil reset as well?

Reply:
Done, reset defaults to legacy.